core/src/main/java/org/infinispan/xsite/SingleXSiteCacheRpcCommand.java

```diff
@@ -0,0 +1,67 @@
+package org.infinispan.xsite;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.commands.remote.CacheRpcCommand;
+import org.infinispan.factories.ComponentRegistry;
+import org.infinispan.util.ByteString;
+
+/**
+ * RPC command to replicate cache rpc operations (i.e. not VisitableCommands) to the backup site.
+ *
+ * @author wburns
+ * @since 12.0
+ */
+public class SingleXSiteCacheRpcCommand extends XSiteReplicateCommand<Object> {
```

Comment:
nitpick! well, if this has the same logic as SingleXSiteRpcCommand probably, it would be better to change the command from VisitableCommand => ReplicableCommand